hw/xbox/nv2a/nv2a_psh.c

```diff
@@ -280,9 +280,10 @@ static QString* get_var(struct PixelShader *ps, int reg, bool is_dest)
         return qstring_from_str("r1");
     case PS_REGISTER_V1R0_SUM:
         add_var_ref(ps, "r0");
-        return qstring_from_str("(v1 + r0)");
+        return qstring_from_str("vec4(v1.rgb + r0.rgb, 0.0)");
```

Comment:
Good catch!

*Note to other reviewers: `PS_REGISTER_V1R0_SUM` refers to the pseudo register listed in the spec as `SPARE0_PLUS_SECONDARY_COLOR_NV` in which "alpha component ... is always zero." [[§3.8.12.3](https://www.khronos.org/registry/OpenGL/extensions/NV/NV_register_combiners.txt)] For historic reasons this file uses Xbox DirectX PSH terminology (mentioned in comment at head of file by espes). Eventually we may want to switch this over to the NVIDIA spec terminology, or at least have some comments describing the name translation, to make cross-referencing with the spec easier.*

The spec also mentions:
> If csc, the color sum clamp, is non-FALSE, the value of SPARE0_PLUS_SECONDARY_COLOR_NV is first clamped to the range [0,1].

And it seems like the code has already been set up for the flag (`ps.final_input.clamp_sum`), it's just not being used:
```
PS_FINALCOMBINERSETTING_CLAMP_SUM=     0x80, // V1+R0 sum clamped to [0,1]
...
ps.final_input.clamp_sum = flags & PS_FINALCOMBINERSETTING_CLAMP_SUM;
```

Do you think we should we also apply clamping at this point while we are here?

Reply:
Yes, see #207 - I have some code which I'll send in the future, it needs some other fixes first.

Reply:
Cool! We can follow up on this with subsequent PR then.